framework/meta/src/folder_structure/relevant_directory.rs

```diff
@@ -33,7 +33,7 @@ pub enum DirectoryType {
 pub struct RelevantDirectory {
     pub path: PathBuf,
     pub version: VersionReq,
-    pub upgrade_in_progress: Option<(&'static str, &'static str)>,
+    pub upgrade_in_progress: Option<(&'static FrameworkVersion, &'static FrameworkVersion)>,
```

Comment:
I think it's possible to drop the `'static` at this point (not sure why it was there in the first place).